engine/src/main/java/org/hibernate/search/util/impl/ScopedAnalyzer.java

```diff
@@ -24,26 +27,28 @@
  */
 public final class ScopedAnalyzer extends AnalyzerWrapper {
```

Comment:
As this class actually *is* a Lucene `Analyzer`, I think the reference should be resolved at this boundary. I.e. this class should ideally remain unchanged, and callers make sure they only invoke it when dealing with Lucene-backed `Analyzer` references.